group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```
@@ -445,6 +446,42 @@ public List<ListGroupsResponseData.ListedGroup> listGroups(List<String> statesFi
         return groupStream.map(group -> group.asListedGroup(committedOffset)).collect(Collectors.toList());
     }
 
+
+    public List<ConsumerGroupDescribeResponseData.DescribedGroup> consumerGroupDescribe(
+        List<String> groupIds,
+        long committedOffset
+    ) {
+        List<ConsumerGroupDescribeResponseData.DescribedGroup> response = new ArrayList<>();
+
+        for (String groupId: groupIds) {
+            Group group = groups.get(groupId, committedOffset);
```

I wonder if we could follow a bit more what we did in `describeGroups`. Would it be possible?

For instance, there we use `genericGroup` which handles the group type check as well. We could use a similar method for consumer groups, I think.

---

Good idea. I added a `consumerGroup` method accordingly and made the structure more consistent with `describeGroups`.

Is it correct for consumer groups to consider the group "Dead" when we can't find a group with the groupId?

---

@riedelmax That's a good question. I think that it would be better to return an error in this case. I would use `GroupIdNotFoundException`.